pyclient/pydeephaven/table.py

```diff
@@ -0,0 +1,683 @@
+                                                               group_by_columns=column_names)
+
+    def median_by(self, column_names: List[str]) -> Table:
+        """ Perform median-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.MEDIAN,
+                                                               group_by_columns=column_names)
+
+    def min_by(self, column_names: List[str]) -> Table:
+        """ Perform min-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.MIN,
+                                                               group_by_columns=column_names)
+
+    def max_by(self, column_names: List[str]) -> Table:
+        """ Perform max-by aggregation on the table and return the result table. Columns not used in the grouping
+        must be of numeric types.
+
+
+        Args:
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.MAX,
+                                                               group_by_columns=column_names)
+
+    def count_by(self, count_column: str, column_names: List[str]) -> Table:
+        """ Perform count-by operation on the table and return the result table. The count of each group is stored in
+        a new column named after the 'count_column' parameter.
+
+
+        Args:
+            count_column (str): the name of the count column
+            column_names (List[str]): the group-by column names
+
+        Returns:
+            a Table object
+
+        Raises:
+            DHError
+
+        """
+        return self.session.table_service.dedicated_aggregator(self, agg_type=AggType.COUNT,
+                                                               group_by_columns=column_names,
+                                                               count_column=count_column)
+
+    def count(self, count_column: str) -> Table:
+        """ Count the number of values in the 'count_column' on the table and return the result in a table with one row
+        and one column.
+
+        Args:
+            count_column (str): the name of the column whose values to be counted
```

Comment:
I don't think this description matches what is actually happening. I *think* `count_column` here is the *output* column name. If you want to count the number of values, I think you use a count distinct.  I think we can get rid of this method...